iothub/device/tests/MessageTests.cs

```diff
@@ -169,66 +169,118 @@ public void SettingMessageAsSecurityMessageTest()
         [TestMethod]
         public void CloneWithBodyTest()
         {
-            using (var original = new Message(Encoding.UTF8.GetBytes("Original copy")))
+            // arrange
+            string contentEncoding = "gzip";
+            string contentType = "text/plain";
+            string userId = "JohnDoe";
+            string propName1 = "test1";
+            string propValue1 = "test_v_1";
+            string propName2 = "test2";
+            string propValue2 = "test_v_2";
+            string originalMessageContent = "Original copy";
+            using var originalMessage = new Message(Encoding.UTF8.GetBytes(originalMessageContent), setDefaultMessageId: true)
             {
-                original.Properties["test1"] = "test_v_1";
-                original.Properties["test2"] = "test_v_2";
-
-                original.ContentEncoding = "gzip";
-                original.ContentType = "text/plain";
-                original.UserId = "JohnDoe";
-
-                using (var clone = original.CloneWithBody(Encoding.UTF8.GetBytes("Cloned version")))
-                { 
-                    Assert.AreEqual("test_v_1", clone.Properties["test1"]);
-                    Assert.AreEqual("test_v_2", clone.Properties["test2"]);
-
-                    Assert.AreEqual("gzip", clone.ContentEncoding);
-                    Assert.AreEqual("text/plain", clone.ContentType);
-                    Assert.AreEqual("JohnDoe", clone.UserId);
-
-                    var clonedContent = default(string);
-
-                    using (var reader = new StreamReader(clone.BodyStream, Encoding.UTF8))
-                    {
-                        clonedContent = reader.ReadToEnd();
-                    }
-                
-                    Assert.AreEqual("Cloned version", clonedContent);
-                }
-            }
+                ContentEncoding = contentEncoding,
+                ContentType = contentType,
+                UserId = userId,
+            };
+            originalMessage.Properties[propName1] = propValue1;
+            originalMessage.Properties[propName2] = propValue2;
+
+            // act
+            string clonedMessageContent = "Cloned version";
+            using var clonedMessage = originalMessage.CloneWithBody(Encoding.UTF8.GetBytes(clonedMessageContent));
+
+            // assert
+            clonedMessage.Properties[propName1].Should().Be(propValue1, "Cloned message should have the original message's properties.");
+            clonedMessage.Properties[propName2].Should().Be(propValue2, "Cloned message should have the original message's properties.");
+            clonedMessage.Properties.Count.Should().Be(2);
```

Comment:
should validate count before dereferencing the dictionary

Reply:
Dereferencing or referencing? It doesn't really matter though, right?

Reply:
dereferencing, meaning we have a a pointer (the dictionary) and we index into it

Reply:
got it